### ISSUE TITLE
auth_oidc basic field mapping not working without local_o365 installed

### ISSUE THREAD

**Issue by nbartley** [microsoft/o365-moodle#1925]
Task: Install auth_oidc in moodle system. Configure against external SSO. Set "description" field to UPN.
Expected result: User Description field contains UPN information.

Result: No change

Install local_o365

Problem goes away, description now contains UPN content.

**Comment by weilai-irl** [comment]
Hi @nbartley,

Thank you for reporting the issue. I acknowledge it's a genuine issue. A fix has been provided in the linked pull request, and will be included in the next release in March 2022 after merge and test.

Regards,
Lai

**Comment by nbartley** [comment]
@weilai-irl Thank for verifying my bug report. 

Would it be possible to remove the pre-identified fields from the mapping and let us manually text enter the token value if not connected to AD/Teams?

**Comment by weilai-irl** [comment]
Hi @nbartley,

Did you mean you want to enter the token claim name to be used in the field mapping? Do you have any particular token claim in mind? Although it's technically possible to allow for text input for claim names, it's extremely prone to errors, therefore we would highly recommend against it.

If you meant something else, please clarify.

Regards,
Lai

**Comment by nbartley** [comment]
I agree that it's prone to errors (LDAP was fun to configure the first
time). Yes, I am asking for the ability to type in claims manually which
allows it to support custom claims in the token. For instance we carry our
own custom claim in our token for student number which makes it easier to
track student data across multiple systems (we use this value in the ID
number field in Moodle)

**Comment by weilai-irl** [comment]
Hi @nbartley,

Please log this as a separate feature request (i.e. a new github issue), so that it can be added to the queue of feature requests.

It's my first time seeing this exact request, so it's possible that it won't get processed in the queue just yet, but if we have enough people voting for it, it will definitely be processed sooner. The queue is constantly reviewed by out team and the Microsoft Education team.

Regards,
Lai
